LochNessBuilder/Builder.cs

```diff
@@ -206,17 +258,12 @@ private IEnumerable<TProp> GetIEnumerableAsAppropriateType<TProp>(IEnumerable<TP
             {
                 if (targetType.IsAssignableFrom(concreteType))
                 {
-                    var valuesInRelevantConcreteType = concreteInitialisers[concreteType](values);
-                    return valuesInRelevantConcreteType;
+                    return concreteInitialisers[concreteType];
                 }
             }
-            
-            throw EnumerableTypeNotSupportedException(targetType);
-        }
 
-        private Exception EnumerableTypeNotSupportedException(Type propType)
-        {
-            throw new NotSupportedException("The IEnumerable handler knows how to create Array, List<>, HashSet<>, Queue<>, Collection<>, ReadOnlyCollection<>, or IQueryable<>. Your property type can't be populated by any of those types, and is thus unsupported by this method. Please use a standard .With() call. PropertyType was :" + propType.ToString());
+            var T = typeof(TProp).Name;
+            throw new NotSupportedException($"Attempted to populate a Property of Type '{targetType.ToString()}'. From the {T} values provided, the IEnumerable handler knows how to create {T}[], List<{T}>, HashSet<{T}>, Queue<{T}>, Collection<{T}>, ReadOnlyCollection<{T}>, or IQueryable<{T}>. Your property type can't be populated by any of those types, and is thus unsupported by this method. Please use a standard .With() call.");
```

Comment:
The recommendation should probably reference `WithFactory` as `With` can't be used for most enumerable types.

Reply:
Good catch. Done.